crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -575,6 +579,21 @@ impl CheckpointStore {
             .verify(committee)
             .map_err(FragmentInternalError::Error)?;
 
+        // Schedule for execution all the certificates that are included here.
```

Comment:
We need a way to know when execution is finished (for epoch change). I guess we can simply look at the pending list in authority_store?

Reply:
This will change in a future PR: we only sign a checkpoint or accept a checkpoint certificate if all transactions included are executed. This will hopefully simplify things.